keanu-project/src/test/java/io/improbable/keanu/algorithms/mcmc/HamiltonianTest.java

```diff
@@ -112,4 +121,30 @@ public void canDefaultToSettingsInBuilderAndIsConfigurableAfterBuilding() {
 
         assertFalse(posteriorSamples.get(A).asList().isEmpty());
     }
+
+    @Test
+    public void canStreamSamples() {
+
+        Hamiltonian hmc = Hamiltonian.builder()
+            .leapFrogCount(10)
+            .stepSize(0.4)
+            .build();
+
+        int sampleCount = 7500;
```

Comment:
It seems unnecessary to introduce another slow test for this:  can you make `sampleCount` small? You might need to change what you're asserting, but what's under test in this method is the ability to stream, not the accuracy of the sampling.